src/cli/config-flags.ts

```diff
@@ -269,8 +269,27 @@ export interface ConfigFlags
     NumberConfigFlags,
     StringNumberConfigFlags,
     LogLevelFlags {
-  task?: TaskCommand | null;
-  args?: string[];
-  knownArgs?: string[];
-  unknownArgs?: string[];
+  task: TaskCommand | null;
+  args: string[];
+  knownArgs: string[];
+  unknownArgs: string[];
 }
+
+/**
+ * Helper function for initializing a `ConfigFlags` object. Provide any overrides
+ * for default values and off you go!
+ *
+ * @param init an object with any overrides for default values
+ * @returns a complete CLI flag object
+ */
+export const createConfigFlags = (init: Partial<ConfigFlags> = {}): ConfigFlags => {
```

Comment:
I wonder if it makes sense to have a separate one of these for creating the config flags for tests.  IDK if reasonable defaults for tests will always be the reasonable defaults we use in `src` 🤔 Thoughts?

Reply:
we could have a separate one for tests if we wanted to, my intention here was to create an object which is the minimum which complies with the interface so I think we'd want _that_ in both test code and, uh, actual code, but if we wanted to have additional defaults for tests we could have a separate function for that, although I think at present it would look something like this:

```ts
const mockConfigFlags = (overrides: Partial<ConfigFlags> = {}): ConfigFlags => {
  const init = createConfigFlags();

  return { ...init, ...overrides };
}
```

Reply:
Yeah - TBH I waffle on what the best thing to do is here. On one hand, it's nice to have a clear distinction between the two - one for testing, one for the impl. On the other, we'd probably end up writing exactly what you have there, which still uses the underlying version for tests.  I say let's let YAGNI ride and keep things as is for now